apps/cow-tools/src/app/milkman/(components)/OrdersTable.tsx

```diff
@@ -109,14 +136,75 @@ function CancelButton({ status }: { status: string }) {
   );
 }
 
+function TransactionInfo({ order }: { order: AllSwapsQuery["swaps"][0] }) {
+  const priceCheckerInfo = getPriceCheckerInfoFromAddressAndChain(
+    order.chainId as 5,
+    order.priceChecker as Address,
+  );
+
+  const expecetedArguments = priceCheckerInfo?.arguments;
+
+  const decodedArgs = priceCheckerInfo
+    ? decodePriceCheckerData(
+        priceCheckerInfo,
+        order.priceCheckerData as `0x${string}`,
+      )
+    : [];
+
+  const priceCheckerUrl = buildBlockExplorerAddressURL({
+    chainId: order.chainId,
+    address: order.priceChecker as Address,
+  });
+
+  return (
+    <div className="text-white">
+      <div className="font-semibold text-2xl flex justify-between">
+        Price Checker Data - {truncateAddress(order.tokenIn?.symbol)} for{" "}
+        {truncateAddress(order.tokenOut?.symbol)}
+      </div>
+      <hr className="mb-2" />
+      <div className="flex flex-col">
+        <div className="flex items-center gap-x-1">
+          <span>
+            Price Checker: {priceCheckerInfo?.name || "Not found"}{" "}
+            {`(${truncateAddress(order.priceChecker as Address)})`}
+          </span>
+          {priceCheckerUrl && (
+            <Link href={priceCheckerUrl.url} target="_blank">
+              <ArrowTopRightIcon className="hover:text-slate11" />
+            </Link>
+          )}
+        </div>
+        {decodedArgs &&
+          expecetedArguments &&
+          decodedArgs.map((argument, index) => (
+            <div key={index}>
+              {expecetedArguments[index].label} :{" "}
+              {expecetedArguments[index].convertOutput(
+                argument,
+                order.tokenOut?.decimals || 18,
+              )}
+            </div>
+          ))}
+        {!decodedArgs.length && (
+          <span>Price Checker Data: Error decoding data</span>
```

Comment:
inline logic like this makes the code a bit harder to parse -- would prefer to have separate functions / components when we have those scenarios to make this more friendly to review